pkg/init/backend/interactive_test.go

```diff
@@ -10,6 +10,78 @@ import (
 	"github.com/redhat-developer/odo/pkg/init/asker"
 )
 
+func TestDetectFramework(t *testing.T) {
+	type args struct {
+		path string
+	}
+	tests := []struct {
+		name             string
+		args             args
+		wantedLanguage   string
+		wantedTools      []string
+		wantedFrameworks []string
+		wantErr          bool
+	}{
+		{
+			name: "Case 1 - Detect Node.JS example",
+			args: args{
+				path: GetTestProjectPath("nodejs"),
+			},
+			wantedLanguage:   "javascript",
+			wantedTools:      []string{"nodejs"},
+			wantedFrameworks: []string{},
+			wantErr:          false,
+		},
+		{
+			name: "Case 2 - Detect java openjdk example",
+			args: args{
+				path: GetTestProjectPath("openjdk"),
+			},
+			wantedLanguage:   "java",
+			wantedTools:      []string{"maven"},
+			wantedFrameworks: []string{},
+			wantErr:          false,
+		},
+		{
+			name: "Case 3 - Detect python example",
+			args: args{
+				path: GetTestProjectPath("python"),
+			},
+			wantedLanguage:   "python",
+			wantedTools:      []string{},
+			wantedFrameworks: []string{},
+			wantErr:          false,
+		},
+		{
+			name: "Case 3 - Detect java wildfly example",
+			args: args{
+				path: GetTestProjectPath("wildfly"),
+			},
+			wantedLanguage:   "java",
+			wantedTools:      []string{},
+			wantedFrameworks: []string{},
+			wantErr:          false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			// Run function DetectFramework
+			languages, err := DetectFramework(tt.args.path)
+
+			if !tt.wantErr == (err != nil) {
+				t.Errorf("unexpected error %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			if !hasWantedLanguage(languages, tt.wantedLanguage, tt.wantedTools, tt.wantedFrameworks) {
```

Comment:
We are mostly testing the `recognizer.Analyze` function here. It would be more important to test how odo behaves depending on hypothetic replies from this `Analyze` function (by mocking it)